### ISSUE TITLE
Cancelled jobs stuck at cancelling

### ISSUE THREAD

**Issue by HansenRene** [ironmansoftware/powershell-universal#3164]
### Version

4.2.13

### Severity

Low

### Steps to Reproduce

Unsure how to reproduce, it happens sporadically, and not on all jobs that are cancelled.

### Expected behavior

```console
When cancelled the jobs should cancel, instead they some times get stuck at "Cancelling" - it seems that they do cancel, as the process ID isn't one that can be found.

I currently have 4 perpetual 'Cancelling' jobs where the oldest is +15 days.
```

### Actual behavior

```console
it don't.
```

### Additional Environment data

experianced on 4.2.12 and 4.2.13.

### Visuals

![image](https://github.com/ironmansoftware/issues/assets/20628942/b79d1d5b-d62a-4c65-b82d-c4b0e2226fff)

**Comment by adamdriscoll** [comment]
This issue has been mentioned on **Ironman Software Forums**. There might be relevant details there:

https://forums.ironmansoftware.com/t/how-to-cancel-job-in-cancelling-state/10381/8

**Comment by insomniacc** [comment]
I'm also seeing this. Just to clarify, a job that I have running on a schedule and has been for a few weeks just got stuck in the 'running' state, no changes have been made to the code for that script, or the configuration of the job/schedule, I've no clue why it just hung (it also prevented future jobs from tiggering), I tried to cancel it and it's now stuck as 'cancelling', even after rebooting the server and updating PSU, nothing i can do will clear it. It's also not showing in hangfire.

![image](https://github.com/ironmansoftware/issues/assets/10046621/907b3015-72c4-4863-8c2c-330ee6bef085)


saw some similar raised older issues that are now closed, wasnt sure if this is linked to these but thought it might be worthwhile noting:
https://github.com/ironmansoftware/issues/issues/3164
https://github.com/ironmansoftware/issues/issues/702
